src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts

```diff
@@ -1167,29 +1418,418 @@ export function generateSamplePoints(
+        const cosTheta = Math.cos(theta);
+        const sinPhi = Math.sin(phi);
+        const cosPhi = Math.cos(phi);
+        const ux = cosTheta * sinPhi;
+        const uy = cosPhi;
+        const uz = sinTheta * sinPhi;
+        out.push([ux * r, uy * r, uz * r]);
+      }
+      break;
+    }
+    case 'cube-edges': {
+      /* eslint-disable prettier/prettier */
+      out.push(
+        // between edges
+        [-1.01, -1.02,  0],
+        [ 1.01, -1.02,  0],
+        [-1.01,  1.02,  0],
+        [ 1.01,  1.02,  0],
+
+        [-1.01,  0, -1.02],
+        [ 1.01,  0, -1.02],
+        [-1.01,  0,  1.02],
+        [ 1.01,  0,  1.02],
+
+        [-1.01, -1.02,  0],
+        [ 1.01, -1.02,  0],
+        [-1.01,  1.02,  0],
+        [ 1.01,  1.02,  0],
+
+        // corners (see comment "Issues with corners of cubemaps")
+        // for why these are commented out.
+        // [-1.01, -1.02, -1.03],
+        // [ 1.01, -1.02, -1.03],
+        // [-1.01,  1.02, -1.03],
+        // [ 1.01,  1.02, -1.03],
+        // [-1.01, -1.02,  1.03],
+        // [ 1.01, -1.02,  1.03],
+        // [-1.01,  1.02,  1.03],
+        // [ 1.01,  1.02,  1.03],
+      );
+      /* eslint-enable prettier/prettier */
+      break;
+    }
+  }
+
+  // Samplers across devices use different methods to interpolate.
+  // Quantizing the texture coordinates seems to hit coords that produce
+  // comparable results to our computed results.
+  // Note: This value works with 8x8 textures. Other sizes have not been tested.
+  // Values that worked for reference:
+  // Win 11, NVidia 2070 Super: 16
+  // Linux, AMD Radeon Pro WX 3200: 256
+  // MacOS, M1 Mac: 256
+  const kSubdivisionsPerTexel = 4;
+  const q = [
+    textureWidth * kSubdivisionsPerTexel,
+    textureWidth * kSubdivisionsPerTexel,
+    6 * kSubdivisionsPerTexel,
+  ];
+  return out.map(c => {
+    const uvw = convertCubeCoordToNormalized3DTextureCoord(c);
+
+    // If this is a corner, move to in so it's not
+    // (see comment "Issues with corners of cubemaps")
+    const ndx = getUnusedCubeCornerSampleIndex(textureWidth, uvw);
+    if (ndx >= 0) {
+      const halfTexel = 0.5 / textureWidth;
+      uvw[0] = clamp(uvw[0], { min: halfTexel, max: 1 - halfTexel });
+    }
+
+    const quantizedUVW = uvw.map((v, i) => {
+      // Quantize to kSubdivisionsPerPixel
+      const v1 = Math.floor(v * q[i]);
+      // If it's nearest and we're on the edge of a texel then move us off the edge
+      // since the edge could choose one texel or another in nearest mode
+      const v2 = nearest && v1 % kSubdivisionsPerTexel === 0 ? v1 + 1 : v1;
+      // Convert back to texture coords
+      return v2 / q[i];
+    }) as vec3;
+    return convertNormalized3DTexCoordToCubeCoord(quantizedUVW);
+  });
+}
+
+function wgslTypeFor(data: number | Dimensionality, type: 'f' | 'i' | 'u'): string {
+  if (Array.isArray(data)) {
     switch (data.length) {
+      case 1:
+        return `${type}f32`;
```

Comment:
seems to produce `ff32`/`if32`/`uf32`

Reply:
fixed (that path isn't taken yet but good catch!)